src/ssl.h

```diff
@@ -31,6 +31,11 @@
 #include <openssl/ssl.h>
 #include <string.h>
 
+#ifdef __APPLE__
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#define ERESTART -1
```

Comment:
Why this? Is it used somewhere?

If this is really needed, please add a small comment to explain.